src/nvidia_rag/utils/summarization.py

```
@@ -0,0 +1,975 @@
+
+    final_summary = current_summaries[0]
+    document.metadata["summary"] = final_summary
+    logger.debug(f"Summary generated for {file_name}: {final_summary[:100]}...")
+
+    return document
+
+
+def _batch_summaries_by_length(
+    summaries: list[str],
+    max_chunk_chars: int,
+    config: NvidiaRAGConfig,
```

Ruff (0.14.5)

Unused function argument: `config`

(ARG001)